src/datasources/person-attribute-datasource.ts

```diff
@@ -0,0 +1,16 @@
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { BaseOpenMRSDataSource } from './data-source';
+
+export class PersonAttributeLocationDataSource extends BaseOpenMRSDataSource {
+  constructor() {
+    super(null);
+  }
+
+  async fetchData(searchTerm: string, config?: Record<string, any>, uuid?: string): Promise<any[]> {
+    const rep = 'v=custom:(uuid,display)';
+    const url = `${restBaseUrl}/location?${rep}`;
+    const { data } = await openmrsFetch(searchTerm ? `${url}&q=${searchTerm}` : url);
+
+    return data?.results;
+  }
```

Comment:
This data source seems to just load locations (not tied down to the "attribute" model). Anything stopping you from reusing the existing [location DS](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/main/src/datasources/location-data-source.ts#L4)`?

Reply:
Done